Floofbot/Services/EventLoggerService.cs

```diff
@@ -0,0 +1,562 @@
+﻿using Discord;
+using Discord.WebSocket;
+using Floofbot.Services.Repository;
+using Serilog;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace Floofbot.Services
+{
+    public class EventLoggerService
+    {
+
+        private DiscordSocketClient _client;
+        public EventLoggerService(DiscordSocketClient client)
+        {
+            _client = client;
+            _client.MessageUpdated += MessageUpdated;
+            _client.MessageDeleted += MessageDeleted;
+            _client.UserBanned += UserBanned;
+            _client.UserUnbanned += UserUnbanned;
+            _client.UserJoined +=UserJoined;
+            _client.UserLeft += UserLeft;
+            _client.GuildMemberUpdated += GuildMemberUpdated;
+        }
+        public async Task<ITextChannel> GetChannel(Discord.IGuild guild, string eventName = null)
+        {
+            if (eventName == null)
+                return null;
+
+            FloofDataContext _floofDb = new FloofDataContext();
+
+            var serverConfig = _floofDb.LogConfigs.Find(guild.Id);
+            System.Reflection.PropertyInfo propertyInfo = serverConfig.GetType().GetProperty(eventName);
+            ulong logChannel = (ulong)(propertyInfo.GetValue(serverConfig, null));
+            var textChannel = await guild.GetTextChannelAsync(logChannel);
+            return textChannel;
+        }
+        public bool IsToggled(IGuild guild)
+        {
+            // check if the logger is toggled on in this server
+            // check the status of logger
+            FloofDataContext _floofDb = new FloofDataContext();
+
+            var ServerConfig = _floofDb.LogConfigs.Find(guild.Id);
+            if (ServerConfig == null) // no entry in DB for server - not configured
+                return false;
+
+            return ServerConfig.IsOn;
+        }
+
+        public Task MessageUpdated(Cacheable<IMessage, ulong> before, SocketMessage after, ISocketMessageChannel chan)
+        {
+            var _ = Task.Run(async () =>
```

Comment:
I'm still not the biggest fan of the usage of `Task.Run` here (and also the return of `Task.CompletedTask`), but will let it go for now. I don't have much input as I'm very much a C# novice, so I'll just put in an issue instead (as you suggested).